app/controllers/admin/api/account/proxy_configs_controller.rb

```diff
@@ -0,0 +1,34 @@
+# frozen_string_literal: true
+
+class Admin::Api::Account::ProxyConfigsController < Admin::Api::BaseController
+  include ApiAuthentication::BySsoToken
+
+  represents :json, entity: ::ProxyConfigRepresenter
+  represents :json, collection: ::ProxyConfigsRepresenter
+
+  clear_respond_to
+  respond_to :json
+
+  rescue_from(ProxyConfig::InvalidEnvironmentError) do
+    render_error 'invalid environment', status: :unprocessable_entity
+  end
+
+  def index
+    respond_with proxy_configs.order(:id).paginate(pagination_params)
+  end
+
+  private
+
+  def proxy_configs
+    accessible_services_ids = System::Database.oracle? ? accessible_services.pluck(:id) : accessible_services.select(:id)
+    ProxyConfig.joins(:proxy).where("proxies.service_id in (?)", accessible_services_ids).by_environment(environment)
```

Comment:
no need to change but it could be also done as: `where(proxies: { id: accessible_services_ids })`